pkgs/development/embedded/arduino/arduino-core/chrootenv.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, buildFHSUserEnv, pname, version, src, withGui ? false, withTeensyduino ? false, gtk3 ? null }:
+let
+  arduino-pkgs = pkgs:
+    let
+      arduino-core = pkgs.callPackage ./core.nix { inherit pname version src withGui withTeensyduino gtk3; };
+    in
+    (with pkgs; [
+      ncurses
+      arduino-core
+      zlib
+      (python3.withPackages (p: with p; [
+        pyserial
+      ]))
+    ]);
+
+in
+buildFHSUserEnv {
+  name = "arduino";
+
+  targetPkgs = arduino-pkgs;
+  multiPkgs = null;
+
+  meta = with lib; {
+    description = "Open-source electronics prototyping platform";
+    homepage = "http://arduino.cc/";
+    license = if withTeensyduino then licenses.unfreeRedistributable else licenses.gpl2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ stunkymonkey ];
+  };
+
+  extraInstallCommands = ''
+    find ${src}
+    echo "###"
+    find $out
+    mkdir -p $out/share/applications
+    # cp ${src}/share/applications/arduino.desktop $out/share/applications/arduino.desktop
```

Comment:
It might help to use `makeDesktopItem`, if the original one doesn't work.

Reply:
i am unable to find it, so yes good idea

Reply:
no idea, why i did not find it yesterday. I just saw it today and fixed the desktop-file with its icon. can you please verify, that it works?

Reply:
I don't have a Linux setup to test this right now. I think that as long as it works for you, and the diff looks good we just merge this PR.